packages/cli/src/services/project-checker.ts

```diff
@@ -69,3 +71,66 @@ export async function checkResourceExists(name: string, placementDir: string, ex
     })
   }
 }
+
+export async function checkCurrentDirBoosterVersion(userAgent: string): Promise<void> {
+  return checkBoosterVersion(userAgent, process.cwd())
+}
+
+async function checkBoosterVersion(userAgent: string, projectPath: string): Promise<void> {
+  const projectVersion = await getBoosterVersion(projectPath)
+  const cliVersion = userAgent.split(' ')[0].split('/')[2]
+  await compareVersionsAndDisplayMessages(cliVersion, projectVersion, projectPath)
+}
+
+async function getBoosterVersion(projectPath: string): Promise<string> {
+  const projectAbsolutePath = path.resolve(projectPath)
+  try {
+    const packageJsonContents = require(path.join(projectAbsolutePath, 'package.json'))
+    const version = packageJsonContents.dependencies['@boostercloud/framework-core']
+    const versionParts = version.replace('^','').replace('.tgz','').split('-')
+    return versionParts[versionParts.length-1]    
+  } catch (e) {
+    throw new Error(
+      `There was an error when recognizing the application. Make sure you are in the root path of a Booster project:\n${e.message}`
+    )
+  }
+}
+
+async function compareVersionsAndDisplayMessages(cliVersion: string, projectVersion: string, projectPath: string): Promise<void> {
+  if (cliVersion === projectVersion)  { return }
+  const cliVersionParts = cliVersion.split('.').map((v) => parseInt(v,10))
+  const projectVersionParts = projectVersion.split('.').map((v) => parseInt(v,10))
```

Comment:
I'd abstract this into a function

```ts
function parseSemver(semverString: string): [number, number, number] {
  return semverString.split('.').map((v) => parseInt(v,10))
}


  const cliVersionParts = parseSemver(cliVersion)
  const projectVersionParts = parseSemver(projectVersion)
```

Reply:
I refactored this by putting all logic in the Semver class. Now comparisons in code looks more semantic